travelpal/scrape-hotels.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/python3
+
+import argparse
+import uuid
+
+from collections import OrderedDict
+from requests import get
+from bs4 import BeautifulSoup
+import html5lib
+
+import psycopg2
+import datetime
+
+
+parser = argparse.ArgumentParser(description='parse destination input')
+parser.add_argument('--dest', action="store", dest='destination')
+parser.add_argument('--store', action="store", dest='shouldStore', default=0)
+
+results = parser.parse_args()
+destination = results.destination
+# 1 = store parsed data
+shouldStoreResults = results.shouldStore
+
+# exit if no argument given
+if destination is None:
+  print("Please enter a desination by adding -dest 'some destination' when running the script")
+  exit()
+
+url = 'https://www.booking.com/searchresults.html?'
+head = {"User-Agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.75 Safari/537.36"}
+
+# use orderedDict to keep order of query parameters
+params = OrderedDict()
+
+params['lang'] = 'en-gb'
+params['sid'] = uuid.uuid4()
+params['src'] = 'index'
+params['src-elem'] = 'sb'
+params['ss'] = destination
+params['checkin_monthday'] = 20
+params['checkin_month'] = 6
+params['checkin_year'] = 2018
+params['checkout_monthday'] = 25
+params['checkout_month'] = 6
+params['checkout_year'] = 2018
+params['group_adults'] = 2
+params['group_children'] = 0
+params['no_rooms'] = 1
+params['from_sf'] = 1
+params['dest_id'] = ''
+params['dest_type'] = ''
+
+response = get(url, headers=head, params=params)
+print(response.status_code)
+
+if(response.status_code != 200):
+    print("Bad response")
+    exit()
+
+# parse response into bs object
+html_soup = BeautifulSoup(response.text, "html5lib")
+
+listing = html_soup.find_all(class_='sr_item')
+hotel_list = []
+base_url = "https://www.booking.com/"
+
+for hotel in listing:
+  name = hotel.select_one("span.sr-hotel__name")
+  district = hotel.select_one(".district_link")
+  price = hotel.select_one(".price")
+  link = hotel.select_one(".hotel_name_link")
+  rating = hotel.select_one(".review-score-badge")  
+
+  if name and price:
+    name = name.text.strip()
+    district = district.text.strip() if district else ""    
+    district = district[0: district.find("Show") - 2].rstrip()
+    
+    # all prices in us currency
+    price = price.text.strip()
+    price = float(price[price.find('$') + 1:].replace(',', ''))
+    link = link.attrs['href']
+    rating = float(rating.text.lstrip().rstrip()) if rating else None    
+ 
+    hotel_info = (name, district, price, base_url + link, rating, datetime.datetime.utcnow(), datetime.datetime.utcnow()) 
+    hotel_list.append(hotel_info)
+
+
+for hotel in hotel_list:
```

Comment:
Probably has negligible effects on the runtime, but just remember to remove this in the final version.